internal/lsp/completions/refs/refs.go

```diff
@@ -0,0 +1,224 @@
+package refs
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-policy-agent/opa/ast"
+
+	rast "github.com/styrainc/regal/internal/ast"
+	"github.com/styrainc/regal/internal/lsp/types"
+)
+
+// ForModule returns a map of refs and details about them to be used in completions that
+// were found in the given module.
+func ForModule(module *ast.Module) map[string]types.Ref {
```

Comment:
The refs returned are for rules and the package itself. Both are used in different providers.